### ISSUE TITLE
Refining bin using Anvi'o

### ISSUE THREAD

**Issue by mdimdadi** [merenlab/anvio#1161]
I was following the tutorial from this page: http://merenlab.org/2015/05/11/anvi-refine/
The first command is: $ anvi-merge */RUNINFO.cp -c contigs.db -o MERGED_PROFILE --skip-hierarchical-clustering

Would you please explain the */RUNINFO.cp here. I was able to make the contigs.db file for -c option.

I have my bins generated from metabat and CONCOCT.

**Comment by meren** [comment]
That is a very old material. That command should go like this now:

```
anvi-merge */PROFILE.db (...)
```

**Comment by mdimdadi** [comment]
ok. What is the PROFILE.db and how to make or generate this? Sorry I might be asking very silly question. I am very new at this.

I just want to refine my bin and all I have now is my bin and Anvi'o installed. Trying to find a good step by step tutorial for refining the bins.

I appreciate your help and time, Murat.  

Imdad

**Comment by ShaiberAlon** [comment]
Hi @mdimdadi,

I would recommend joining our discussion group on Slack, here is a link that would allow you to join:
https://slackin-ezbpfhwsmh.now.sh/

Specifically, there was recently a relevant discussion there: https://anvio.slack.com/archives/C8SFMGYF3/p1556647192071600

Please let me know if that was helpful.

**Comment by meren** [comment]
And this is another discussion that could help:

https://anvio.slack.com/archives/C8SFMGYF3/p1555984584018500

A screenshot from that discussion; you need to replace 'for each isolate' down below with 'for each MAG' for your case:

![image](https://user-images.githubusercontent.com/197307/57704972-8db92000-7628-11e9-85be-d6c89b605b32.png)

**Comment by mdimdadi** [comment]
I have done the following so far following your suggestion and got the error provided down below:
sorry for the long post though. I really appreciate your time and help.

**$ anvi-gen-contigs-database -f ../../567_cont3.fa -o contigs.db**
this created a contigs.db file from my contigs.fa file
the I ran: 
**$ anvi-init-bam ../../567_cont3_aln-pe_sorted.bam -o 567_cont3_aln-pe_sorted_indexed.bam**
this indexed my .bam file
then I ran:
**$ anvi-profile -i 567_cont3_aln-pe_sorted_indexed.bam -c contigs.db -T 32**
which created a PROFILE.db file
then I am trying to run:
**$ anvi-interactive -p 567_cont3_aln-pe_sorted_indexed.bam-ANVIO_PROFILE/PROFILE.db –c 567_cont3.db**
which is giving me following error:

""""WARNING
===============================================
This single profile database does not seem to have any hierarchical clustering
that is required by the interactive interface. You must use `--cluster-contigs`
flag for single profiles if you would like to see a hierarchical clustering
dendrogram in the center of your display. Please read the help menu for anvi-
profile, and/or refer to the tutorial.

Config Error: Apologies :( The code managed to came all the way down here without any item
              orders set for your project, which makes it impossible for anvi'o intearctive
              interface to display anything :/ Please check warning messages you may have on
              your screen.""""""

**Comment by ShaiberAlon** [comment]
Hi,

In order to visualize a single profile database you must redo the anvi-profile, but add the flag 
 --cluster-contigs (As is mentioned in the warning).

In other words, redo the profile step like this:
anvi-profile -i 567_cont3_aln-pe_sorted_indexed.bam -c contigs.db -T 32 --cluster-contigs

**Comment by mdimdadi** [comment]
Oh. Thank you very much. I have started it already. I will let you know how it goes.

**Comment by mdimdadi** [comment]
I have run the anvi-interactive with --server-only option and got the following messages:
 * The server is now listening the port number "8082". When you are finished, press
CTRL+C to terminate the server.

when I go to a browser to see the visualization and paste the link "http://localhost:8082" nothing shows.

What should I do?

Thanks,
Imdad

**Comment by meren** [comment]
Are you running `anvi-interactive` this on a server through SSH or on your own computer?

If you are running it on your own computer, restart your computer, and try again.

If you are running it on a server, take a look at this blog post:

http://merenlab.org/2015/11/28/visualizing-from-a-server/

**Comment by mdimdadi** [comment]
Yes I am working on a server from a remote connection. 

This is what I have done:
connected to my server rameshgoel.civil.utah.edu using 
$ ssh -L 22:localhost:8095 mdimdadi@155.98.14.181
I have used 22 as this is our port number for our server and forwarded it to localhost server port 8095.
then I have executed "anvi-interactive" command and got the following message:
* The server is now listening the port number "8095". When you are finished, press
CTRL+C to terminate the server.

Next, I opened a browser and typed the address http://localhost:8095 but got nothing. It said:
This site can’t be reached.

Sorry to bother again and again.
